src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -178,6 +188,60 @@ public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String d
         }
     }
 
+    public DRSAccessURL getAccessUrlForObjectId(AuthenticatedUserRequest authUser, String objectId, String accessId) {
+        DRSObject object = lookupObjectByDrsId(authUser, objectId, false);
+
+        DrsId drsId = drsIdService.fromObjectId(objectId);
+        Snapshot snapshot = snapshotService.retrieve(UUID.fromString(drsId.getSnapshotId()));
+        SnapshotProject snapshotProject = snapshotService.retrieveAvailableSnapshotProject(snapshot.getId());
+        GoogleProjectResource projectResource = snapshot.getProjectResource();
+
+        FSItem fsObject = null;
```

Comment:
The code doesn't need to initialize `fsObject`, and it can be declared `final`, since it will be initialized before its use.
```suggestion
        final FSItem fsObject;
```

Reply:
I'm not sure that you even need all of this FSItem code...doesn't that all live inside of the DrsObjec that you get in the beginning?  I'd expect that you'd just iterate over its access methods and if it finds an access method with the access id, sign the URL.  You may also want to preemptively do a check so make sure that this snapshot is a GCP-backed snapshot